packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -890,7 +942,11 @@ class MediaQuery extends InheritedWidget {
   static MediaQueryData of(BuildContext context) {
     assert(context != null);
     assert(debugCheckHasMediaQuery(context));
-    return context.dependOnInheritedWidgetOfExactType<MediaQuery>()!.data;
+    return _of(context);
+  }
+
+  static MediaQueryData _of(BuildContext context, [MediaQueryAspect? aspect]) {
+    return InheritedModel.inheritFrom<MediaQuery>(context, aspect: aspect)!.data;
```

Comment:
Let's move the `assert(debugCheckHasMediaQuery(context));` into this method so people using the more specific of methods get a better error message.

Reply:
Please also check whether the error message in debugCheckHasMediaQuery needs updating.